### PR TITLE
improve: Home button in notebook menu

### PR DESCRIPTION
**Pull request by wasimsandhu**
## 📝 Summary

Completes #1984. Adds home button at top left of open notebook page next to app config and shutdown buttons.

Also updates "Running notebooks" section on home page to open running notebooks in the same tab for navigation consistency.

## 🔍 Description of Changes

Screenshot 2024-08-10 at 3 30 45 PM

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [x] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://discord.gg/JE7nhX6mD8), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers

@mscolnick

### CODE REVIEW

File: frontend/src/components/editor/controls/Controls.tsx [hunk 1]
```diff
@@ -89,6 +90,7 @@ export const Controls = ({
       {!closed && (
         <div className={topRightControls}>
           {presenting && <LayoutSelect />}
+          <HomeButton />
```
Comment (akshayka) [inline frontend/src/components/editor/controls/Controls.tsx:93]: Nit: I suggest putting `HomeButton` after `ConfigButton`.

**Review by akshayka — COMMENTED** [review]
Thanks for the PR! Two pieces of feedback.

1. The home button is styled differently than the NotebookMenuDropdown and Config buttons — can you update the style of the home button to match the NotebookMenuDropdown and Config buttons?

2. When a notebook is started with `marimo new` or `marimo edit notebook.py`, we don't serve the home page — in which case the home button is just refreshing the notebook, which is confusing. I suggest only showing the home button when we're actually serving the home page.

**Comment by wasimsandhu** [comment]
Ah I didn't account for `marimo new` and `marimo edit filename.py`. Is there some flag I can use somewhere, or would it be okay to infer that a home page wasn't rendered by checking whether the URL has query parameters?

**Review by akshayka — COMMENTED** [review]
Iterating on the design — I think it makes most sense for the home to be an action in the hamburger menu, right above "restart notebook", with title "Return home". Can you make that change?

image

**Comment by wasimsandhu** [comment]
Sure! Personally I think a navigation item might get lost in a context menu, but I'm not particularly opinionated about this

**Comment by akshayka** [comment]
> Sure! Personally I think a navigation item might get lost in the hamburger menu (would a user intuitively open a menu to return home?), but I'm not particularly opinionated about this

The home action I view as a convenience, not an integral part of the top-bar, so I'd like to emphasize it less.

**Comment by wasimsandhu** [comment]
Screenshot 2024-08-11 at 10 08 02 AM
🙂

**Comment by akshayka** [comment]
> Screenshot 2024-08-11 at 10 08 02 AM
> 🙂

Fantastic thank you!

File: frontend/src/components/editor/actions/useNotebookActions.tsx [hunk 2]
```diff
@@ -299,6 +300,15 @@ export function useNotebookActions() {
 
     {
       divider: true,
+      icon: <Home size={14} strokeWidth={1.5} />,
+      label: "Return home",
+      hidden: !location.search.includes("file"),
```
Comment (akshayka) [inline frontend/src/components/editor/actions/useNotebookActions.tsx:305]: @mscolnick can you comment whether this is an acceptable way to check whether the server is capable of serving the "home" page?

**Comment by wasimsandhu** [comment]
Seems as though a test is failing with Python 3.11, is it just a fluke?

**Comment by akshayka** [comment]
> Seems as though a test is failing with Python 3.11, is it just a fluke?

Yea don't worry about the failure, it's unrelated, I'm handling it in another PR.

File: frontend/src/components/editor/actions/useNotebookActions.tsx (hunk 2 above)
Comment (mscolnick) [reply frontend/src/components/editor/actions/useNotebookActions.tsx:305]: Yea this is fine. Potentially in a future change, we should allow going back home regardless, and just use the current directly for "marimo new"

**Review by akshayka — APPROVED** [review]
Thank you!

**Pull request merged**